commit 052c2005b6ecedc5abad86632f5781adda310aa7
Author: Matthias Reichl <hias@horus.com>
Date:   Sun Oct 11 12:28:30 2015 +0200

    bcm2835-dma: Limit cyclic transfers on lite channels to 32k
    
    Transfers larger than 32k cause repeated clicking with I2S soundcards.
    The exact reason is yet unknown, so limit to 32k as bcm2708-dmaengine
    did as an intermediate fix.

diff --git a/drivers/dma/bcm2835-dma.c b/drivers/dma/bcm2835-dma.c
index fe1fd60..0adc347 100644
--- a/drivers/dma/bcm2835-dma.c
+++ b/drivers/dma/bcm2835-dma.c
@@ -144,6 +144,12 @@ struct bcm2835_desc {
  */
 #define MAX_LITE_TRANSFER	(SZ_64K - 4)
 
+/*
+ * Transfers larger than 32k cause issues with the bcm2708-i2s driver,
+ * so limit transfer size to 32k as bcm2708-dmaengine did.
+ */
+#define MAX_CYCLIC_LITE_TRANSFER	SZ_32K
+
 static inline struct bcm2835_dmadev *to_bcm2835_dma_dev(struct dma_device *d)
 {
 	return container_of(d, struct bcm2835_dmadev, ddev);
@@ -412,7 +418,7 @@ static struct dma_async_tx_descriptor *bcm2835_dma_prep_dma_cyclic(
 	d->c = c;
 	d->dir = direction;
 	if (c->ch >= 8) /* LITE channel */
-		max_size = MAX_LITE_TRANSFER;
+		max_size = MAX_CYCLIC_LITE_TRANSFER;
 	else
 		max_size = MAX_NORMAL_TRANSFER;
 	period_len = min(period_len, max_size);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1023_052c2005b6ecedc5abad86632f5781adda310aa7.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1023_052c2005b6ecedc5abad86632f5781adda310aa7.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
